src/main/java/seedu/address/logic/parser/SearchCommandParser.java

```diff
@@ -0,0 +1,139 @@
+        List<String> prefixes = keywordsAndPrefixes.getKey();
+        List<List<String>> keywords = keywordsAndPrefixes.getValue();
+        return new SearchCommand(new ContactContainsAnyKeywordsPredicate(prefixes, keywords));
+    }
+
+    private SearchCommand parseSearchWithEmptyCondition(ArgumentMultimap argMultimap) {
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_NAME.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_NAME)));
+        } else if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_PHONE.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_PHONE)));
+        } else if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_EMAIL.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_EMAIL)));
+        } else if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_ADDRESS.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_ADDRESS)));
+        } else {
+            return new SearchCommand(new ContactContainsKeywordsPredicate(PREFIX_TAG.getPrefix(),
+                    argMultimap.getAllValues(PREFIX_TAG)));
+        }
+    }
+
+    private Pair<List<String>, List<List<String>>> extractPrefixesAndKeywords(ArgumentMultimap argMultimap)
+            throws ParseException {
+        List<String> prefixes = new ArrayList<>();
+        List<List<String>> keywords = new ArrayList<>();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            prefixes.add(PREFIX_NAME.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_NAME));
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            prefixes.add(PREFIX_PHONE.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_PHONE));
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            prefixes.add(PREFIX_EMAIL.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_EMAIL));
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            prefixes.add(PREFIX_ADDRESS.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_ADDRESS));
+        }
+        if (argMultimap.getValue(PREFIX_TAG).isPresent()) {
+            prefixes.add(PREFIX_TAG.getPrefix());
+            keywords.add(argMultimap.getAllValues(PREFIX_TAG));
+        }
+        if (prefixes.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SearchCommand.MESSAGE_USAGE));
+        }
+        return new Pair<>(prefixes, keywords);
+    }
```

Comment:
Actually this is stemming from the `switch-case` above. Initially I tried implementing it using `switch-case` but then realized that `switch` can only takes in `constant expression`, like `String`, `int`, etc. but not `Prefix`. So that's why I constructed this to extract the `String` type of the `prefix`. But then right now it's either I stick to `if-else` using `Prefix` or `switch-case` using `String`.